### PR TITLE
Feature/view profile history

### PR DESCRIPTION
**Pull request by yazhyvyi**

### CODE REVIEW

File: client/.gitignore [hunk 1]
```diff
@@ -20,3 +20,4 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 .eslintrc
+package-lock.json
```
Comment (Merteg) [inline client/.gitignore]: Why do you add this file to gitignore?

File: client/src/components/EventPage/index.jsx [hunk 2]
```diff
@@ -37,6 +42,49 @@ class EventPage extends Component {
...
+          'if_subbed': false,
+          'event_id': this.props.match.params.id,
+          'user_id': sessionStorage['User-id']
+        }
+      }
+      else {
+        data = {
+          'if_subbed': true,
+          'event_id': this.props.match.params.id,
+          'user_id': sessionStorage['User-id']
```
Comment (Merteg) [inline client/src/components/EventPage/index.jsx]: You don't need to duplicate 'event_id' and 'user_id' in this `if`, you can assign a value to it outside of `if` statement.

File: client/src/components/EventPage/index.jsx [hunk 3]
```diff
@@ -37,6 +42,49 @@ class EventPage extends Component {
...
+        }
+      }
+
+      request('/subscribe/' + sessionStorage['User-id'], "POST",
+              JSON.stringify(data))
+      .then(data => {
+        if (data.success) {
+          if (data.is_subbed) {
```
Comment (Merteg) [inline client/src/components/EventPage/index.jsx]: You can unite this `if`'s.

File: client/src/components/EventPage/index.jsx [hunk 4]
```diff
@@ -53,12 +101,54 @@ class EventPage extends Component {
         this.setState({
           category: data.category,
           tags: data.tags,
-          status_str: data.status_str,
         });
+
+        this.setState({
```
Comment (Merteg) [inline client/src/components/EventPage/index.jsx]: Also, you don't need to separate this setState and previous.

File: client/src/components/EventPage/index.jsx [hunk 5]
```diff
@@ -53,12 +101,54 @@ class EventPage extends Component {
...
       };
     });
   }
 
+  renderSubscribedUsers() {
+    if (this.state.any_subs) {
+      return (
+        <React.Fragment>
+        {this.state.subs.map((element) => {
+          let avatar = element.avatar === null ? "/person.jpg" : element.avatar
```
Comment (Merteg) [inline client/src/components/EventPage/index.jsx]: When you don't change a variable, better to use `const`. [Style Guide](https://github.com/airbnb/javascript/#references--prefer-const)

File: client/src/components/EventPage/index.jsx [hunk 6]
```diff
@@ -37,6 +42,49 @@ class EventPage extends Component {
     }
   }
 
+  SubscribeEvent = (eve) => {
```
Comment (Merteg) [inline client/src/components/EventPage/index.jsx]: Hmm, `subscribeEvent` seems better))
[Guide](https://github.com/airbnb/javascript/#naming--camelCase)

File: client/src/components/Profile.jsx [hunk 7]
```diff
@@ -1,20 +1,61 @@
...
 
     componentDidMount() {
         request('/profile/' + this.props.match.params.profile_id)
         .then(data => {
-            this.setState({ user: data });
+          this.setState({ user: data });
         })
+        request('/profile-history/' + this.props.match.params.profile_id)
+        .then(data => {
+          console.log("DATA: " + data)
```
Comment (Merteg) [inline client/src/components/Profile.jsx]: Remove console.log. All console.log's

File: client/src/components/Profile.jsx [hunk 8]
```diff
@@ -1,20 +1,61 @@
...
+              history: data.history,
+              events: data.events,
+              any_history: true
+            })
+            if (data.num_of_events === 1) {
+              this.setState({
+                display_method: 12,
+                card_size: 'large'
+             })
+              this.setState({  })
```
Comment (Merteg) [inline client/src/components/Profile.jsx]: Remove this.

File: client/src/components/Profile.jsx [hunk 9]
```diff
@@ -49,46 +90,97 @@ class Profile extends Component {
     renderEditButton = () => {
         if (this.props.match.params.profile_id === sessionStorage['User-id']) {
             return (
-                <Link
-                  className="waves-effect waves-light btn"
-                  to={"/profile-edit/" + sessionStorage['User-id']}>
+                <a
```
Comment (Merteg) [inline client/src/components/Profile.jsx]: Are you sure that the `Link` doesn't work here?

File: client/src/components/Profile.jsx [hunk 10]
```diff
@@ -49,46 +90,97 @@ class Profile extends Component {
...
+      return (
+        <React.Fragment>
+        {this.state.events.map((element) => {
+          return (
+            <Col key={element.id}>
+              <Card
+                horizontal
+                header={
+                  <CardTitle
+                    image={
```
Comment (Merteg) [inline client/src/components/Profile.jsx]: You don't need to add line break here

File: client/src/components/Profile.jsx [hunk 11]
```diff
@@ -49,46 +90,97 @@ class Profile extends Component {
...
-              </Col>
-            </Row>
-            <br />
+          <Card
+            className="profile-card"
+            horizontal
+            header={
+              <CardTitle
+                image={
+                  this.state.user.avatar ===
```
Comment (Merteg) [inline client/src/components/Profile.jsx]: Don't separate statement like this. [Guide](https://github.com/airbnb/javascript/#control-statements)

File: client/src/components/Profile.jsx [hunk 12]
```diff
@@ -49,46 +90,97 @@ class Profile extends Component {
...
+                  null ? "/default_profile.png" : this.state.user.avatar
+                }
+              >
+              </CardTitle>
+            }
+            actions={
+              [ <React.Fragment>{this.renderEditButton()}</React.Fragment> ]
+            }
+          >
+            <h4 className="title-profile">Profile&nbsp;&nbsp;<span                  className="profile-title-nick">@{this.state.user.nickname}</span>
```
Comment (Merteg) [inline client/src/components/Profile.jsx]: Many many useless white spaces.

File: server/server/views/event.py [hunk 13]
```diff
@@ -128,8 +130,22 @@ def get(self):
         }
         if request.user is not None and event_obj.author_id == request.user.id:
             response['status'] = model_to_dict(self.event_history)
-        response['status_str'] = self.event_status
-
+            response['status_str'] = self.event_status
```
Comment (Merteg) [inline server/server/views/event.py]: This expression must be outside of if statement. This is very important.

File: server/server/views/profile-history.py [hunk 14]
```diff
@@ -0,0 +1,41 @@
+from cornice.resource import Service
```
Comment (Merteg) [inline server/server/views/profile-history.py:1]: This file name does not meet the PEP8 ( I talk about profile-history.py) [PEP8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names)

File: server/server/views/profile-history.py [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+    id_arr = [obj.id for obj in events]
+    response['num_of_events'] = len(id_arr)
+    history = EventHistory.\
+                    get_event_history_by_user_id(request, id_arr=id_arr)
+    history_dict = [model_to_dict(obj) for obj in history]
+    events_dict = [model_to_dict(obj) for obj in events]
+
+    response['events'] = events_dict
+    response['history'] = history_dict
+    response['success'] = True
```
Comment (Merteg) [inline server/server/views/profile-history.py]: You don't need to separate these expressions.

**Review by Merteg — CHANGES_REQUESTED** [review]
image
Please, don't add images like this, we also have theirs in other, nested, folder.

**Comment by Merteg** [comment]
image
Please resolve these warnings. 
image
I like this design, really, but there is a small (or not) problem with images.
image
Also, there are many warnings in the console, at profile page, resolve it.

**Comment by Merteg** [comment]
Also, unsubscribe don't work immediately.
I mean, when I click on `Subscribe` it's changing to `Unsubscribe`, but when I click on `Unsubscribe` it's not.
image
I have a 1366x768 screen resolution, and it seems bad on this. Can you resolve it?

**Comment by yazhyvyi** [comment]
Update

File: client/src/components/EventPage/index.jsx [hunk 16]
```diff
@@ -42,40 +42,39 @@ class EventPage extends Component {
...
 
-  SubscribeEvent = (eve) => {
+  subscribeEvent = (eve) => {
     eve.preventDefault()
 
-    let data;
+    let data = {
+      'event_id': this.props.match.params.id,
+      'user_id': sessionStorage['User-id'],
+      'if_subbed': null
```
Comment (Merteg) [inline client/src/components/EventPage/index.jsx]: You can set 'if_subbed' true by default. After this, you don't need an else statement.

File: client/src/components/EventPage/index.jsx [hunk 17]
```diff
@@ -130,7 +126,8 @@ class EventPage extends Component {
       return (
         <React.Fragment>
         {this.state.subs.map((element) => {
-          let avatar = element.avatar === null ? "/person.jpg" : element.avatar
+          const avatar = element.avatar ===
+                         null ? "img/person.jpg" : element.avatar
```
Comment (Merteg) [inline client/src/components/EventPage/index.jsx]: Why do you add a line break?

**Review by Merteg — COMMENTED** [review]
Also, resolve merge conflicts.

**Comment by yazhyvyi** [comment]
Update

**Review by Merteg — APPROVED** [review]

**Review by RayMet — APPROVED** [review]

**Pull request merged**